subsys/power/policy.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <soc.h>
+#include "pm_policy.h"
+
+#define LOG_MODULE_NAME power
+#define LOG_LEVEL CONFIG_PM_LOG_LEVEL /* From power module Kconfig */
+#include <logging/log.h>
+LOG_MODULE_DECLARE();
+
+#ifdef CONFIG_TICKLESS_KERNEL
+#define SECS_TO_TICKS		CONFIG_TICKLESS_KERNEL_TIME_UNIT_IN_MICRO_SECS
+#else
+#define SECS_TO_TICKS		CONFIG_SYS_CLOCK_TICKS_PER_SEC
```

Comment:
Such tick <-> time conversion is imprecise on some SoCs (expecially nRF5x). If you need a more precise timing, please use `__ticks_to_ms()` and `_ms_to_ticks()`.